ground/src/main/java/com/google/android/ground/ui/MarkerIconFactory.kt

```diff
@@ -30,30 +31,43 @@ import javax.inject.Singleton
 
 @Singleton
 class MarkerIconFactory @Inject constructor(@ApplicationContext private val context: Context) {
-
-  fun getMarkerBitmap(color: Int, currentZoomLevel: Float, isSelected: Boolean = false): Bitmap {
-    val outline = AppCompatResources.getDrawable(context, R.drawable.ic_marker_outline)
-    val fill = AppCompatResources.getDrawable(context, R.drawable.ic_marker_fill)
-    val overlay = AppCompatResources.getDrawable(context, R.drawable.ic_marker_overlay)
+  /** Create a scaled bitmap based on the dimensions of a given [Drawable]. */
+  private fun createBitmap(
+    drawable: Drawable,
+    zoomLevel: Float,
+    isSelected: Boolean = false
+  ): Bitmap {
     // TODO: Adjust size based on selection state.
     var scale = ResourcesCompat.getFloat(context.resources, R.dimen.marker_bitmap_default_scale)
-    if (currentZoomLevel >= HomeScreenMapContainerViewModel.ZOOM_LEVEL_THRESHOLD) {
+
+    if (zoomLevel >= HomeScreenMapContainerViewModel.ZOOM_LEVEL_THRESHOLD) {
```

Comment:
Can we move this constant into this file or pass it in as an arg to avoid circular references and enforce separation of concerns here? 

Also could you please add a comment somewhere explaining what this does?

Reply:
Done and done.